tests/APIv0/SmokeTest.php

```diff
@@ -365,6 +376,121 @@ public function testPostRestrictedDiscussion() {
         );
     }
 
+    /**
+     * Test saving a draft.
+     *
+     * @depends testRegisterBasic
+     * @large
+     */
+    public function testSaveDraft() {
+        $api = $this->api();
+        $api->setUser($this->getTestUser());
+
+        $draft = [
+            'CategoryID' => 1,
+            'Name' => 'Draft Test',
+            'Body' => 'Test posting a new draft',
+            'Save Draft' => 'Save Draft'
+        ];
+
+        $r = $api->post(
+            '/post/discussion.json',
+            $draft
+        );
+        $statusCode = $r->getStatusCode();
+        $this->assertEquals(200, $statusCode);
+
+        $draftModel = new \DraftModel();
+        $postedDraft = $draftModel->getWhere(['Name' => $draft['Name']])->firstRow(DATASET_TYPE_ARRAY);
```

Comment:
Doesn't DraftID come over as part of the response (`$r`) when saving a draft? Could that be used here, instead of trying to match on the name?

Reply:
We don't get a draftID in the response, it's very limited.  

$r->getBody returns
{
"isHomepage" : false,
"Type" : "Discussion"
}

Reply:
That's unusual. Do you have any idea why? When saving a draft on the frontend, you get the following response:

```json
{
    "DiscussionID": "",
    "DraftID": 2,
    "FormSaved": true,
    "DeliveryType": "VIEW",
    "Data": "[...]",
    "InformMessages": [
        {
            "CssClass": "Dismissable AutoDismiss",
            "Message": "Draft saved at 11:00AM"
        }
    ],
    "ErrorMessages": "",
    "RedirectTo": "",
    "RedirectUrl": ""
}
```

Reply:
I was missing some params in my request.